include/nuttx/sched.h

```diff
@@ -470,9 +429,7 @@ struct task_group_s
 
   /* [at|on]exit support ****************************************************/
 
-#ifdef CONFIG_SCHED_EXIT_MAX
-  struct exitinfo_s tg_exit[CONFIG_SCHED_EXIT_MAX];
-#endif
+  FAR struct atexit_list_s *tg_exit;
```

Comment:
it's better to move task_info_s instead

Reply:
> I don't agree on this. In my opinion the group structure is the right place. TLS is thread local storage, right ? And every thread has its own?

There are two structure in the userspace: tls_info_s and task_info_s:

1. tls_info_s is the userspace counterpart of tcb_s
2. task_info_s is the userspace counterpart of task_group_s

> 
> The exit functions should be on process level, and they are called when the process exits, not when a thread exits.

Yes, so I suggest to put into task_info_s not tls_info_s

Reply:
Yes task_info_s is much better, thanks for pointing that out.

This is now done